lost/src/main/java/com/mapzen/android/lost/internal/LocationSettingsResultRequest.java

```diff
@@ -0,0 +1,218 @@
+package com.mapzen.android.lost.internal;
+
+import com.mapzen.android.lost.api.LocationRequest;
+import com.mapzen.android.lost.api.LocationSettingsRequest;
+import com.mapzen.android.lost.api.LocationSettingsResult;
+import com.mapzen.android.lost.api.LocationSettingsStates;
+import com.mapzen.android.lost.api.PendingResult;
+import com.mapzen.android.lost.api.ResultCallback;
+import com.mapzen.android.lost.api.Status;
+
+import android.app.PendingIntent;
+import android.bluetooth.BluetoothAdapter;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.location.LocationManager;
+import android.os.AsyncTask;
+import android.os.Handler;
+import android.support.annotation.NonNull;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static com.mapzen.android.lost.api.Status.CANCELLED;
+import static com.mapzen.android.lost.api.Status.INTERNAL_ERROR;
+import static com.mapzen.android.lost.api.Status.INTERRUPTED;
+import static com.mapzen.android.lost.api.Status.RESOLUTION_REQUIRED;
+import static com.mapzen.android.lost.api.Status.SETTINGS_CHANGE_UNAVAILABLE;
+import static com.mapzen.android.lost.api.Status.SUCCESS;
+import static com.mapzen.android.lost.api.Status.TIMEOUT;
+
+public class LocationSettingsResultRequest extends PendingResult<LocationSettingsResult> {
```

Comment:
After reading and testing the code this class certainly seems like it works as intended. Nicely done.

However I'm still hesitant to introduce this level of concurrency into the the framework for what are essentially short-lived operations (`isProviderEnabled()`, `hasSystemFeature()`, etc) that could be executed on the UI thread with no real drawbacks.

Here are some questions that come to mind:

1. What benefits do we get from wrapping the `FutureTask` in an `AsyncTask`? Could we just use a `FutureTask` directly?
2. Is it possible to implement a version of the `PendingResult` API without the actual use of concurrency? Can we "fake" it until we actually have a need for it?
3. Should we eliminate the `PendingResult` API all together? In our quest to achieve API parity with play services are we introducing code/features that don't really make sense in the context of the LOST framework?

Play services has a ton of overhead due to authentication and IPC that we don't have to deal with (at least not yet) which probably makes more sense for asynchronous operations.

Concurrency can introduce subtle bugs in the code that can be hard to track down. It also makes unit testing more challenging. So before we commit to this path forward I just want to make sure we are considering all our options and the pros/cons of each.